source/renderer/app/containers/MenuUpdater/useMenuUpdater.spec.js

```diff
@@ -38,7 +40,15 @@ const makeRebuildApplicationMenu = ({
   send = () => {},
 } = {}): RebuildApplicationMenu => ({ send }: any);
 
-const defaultArgs: MakeReactionCallbackArgs = {
+const renderComponent = (args) => {
+  const Component = () => {
```

Comment:
Another suggestion is that you could test the `MenuUpdater` to have a better integration test. Currently you are only testing the internal implementation. 

```js
function renderComponent(spy) {
return <>
<MenuUpdaterProvider appStore={{}} profileStore={{}} router={{}} stakingStore={{}} uiDialogsStore={{}}>
 <MenuUpdater onUpdate={({isNavigationEnabled, walletSettingsState}) => spy({isNavigationEnabled, walletSettingsState})} />
</MenuUpdaterProvider>
}
```

Reply:
The `MenuUpdater` component itself is not important as its role is only to provide required data using the mobx api. The sole logic is contained in the useMenuUpdater. Testing the MenuUpdater would require from me to involve mobx to testing which IMO is not worth for value it would give. I think we don't need to test things like connecting the component to mobx store